src/components/organisms/CreateProjectModal/CreateProjectModal.tsx

```diff
@@ -0,0 +1,131 @@
+import {useEffect, useState} from 'react';
+
+import {Button, Form, Input, Modal} from 'antd';
+import {useForm} from 'antd/lib/form/Form';
+
+import {useAppDispatch, useAppSelector} from '@redux/hooks';
+import {setCreateProject} from '@redux/reducers/appConfig';
+import {closeCreateProjectModal} from '@redux/reducers/ui';
+
+import FileExplorer from '@components/atoms/FileExplorer';
+
+import {useFileExplorer} from '@hooks/useFileExplorer';
+
+import {useFocus} from '@utils/hooks';
+
+import Colors from '@styles/Colors';
+
+const CreateProjectModal: React.FC = () => {
+  const uiState = useAppSelector(state => state.ui.createProjectModal);
+  const dispatch = useAppDispatch();
+  const [createProjectForm] = useForm();
+  const [inputRef, focus] = useFocus<any>();
+  const [formStep, setFormStep] = useState(1);
+  const [formValues, setFormValues] = useState({projectName: '', location: ''});
+
+  const {openFileExplorer, fileExplorerProps} = useFileExplorer(
+    ({folderPath}) => {
+      if (folderPath) {
+        createProjectForm.setFieldsValue({location: folderPath});
+      }
+    },
+    {isDirectoryExplorer: true}
+  );
+
+  const onFinish = (values: {projectName: string; location: string}) => {
+    setFormValues(values);
+    if (uiState.fromTemplate && formStep === 1) {
+      setFormStep(2);
+    }
+  };
+
+  useEffect(() => {
+    console.log('formValues', formValues);
+    if (!uiState.fromTemplate && formValues.location && formValues.projectName) {
+      dispatch(setCreateProject({rootFolder: formValues.location, name: formValues.projectName}));
+      closeModal();
+    }
+  }, [formValues]);
+
+  useEffect(() => {
+    if (uiState.isOpen) {
+      focus();
+    } else {
+      createProjectForm.resetFields();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [uiState]);
```

Comment:
do you need this to be called every time the entire `uiState` object is changed?